01-import_and_maxfilter.py

```diff
@@ -219,63 +233,114 @@ def run_maxwell_filter(subject, session=None):
+                logger.info(gen_log_message(message=msg, step=1,
+                                            subject=subject, session=session))
+
+            # Keyword arguments shared between Maxwell filtering of the
+            # experimental and the empty-room data.
+            common_mf_kws = dict(calibration=config.mf_cal_fname,
+                                 cross_talk=config.mf_ctc_fname,
+                                 st_duration=config.mf_st_duration,
+                                 origin=config.mf_head_origin,
+                                 coord_frame='head',
+                                 destination=dev_head_t)
+
+            raw_sss = mne.preprocessing.maxwell_filter(raw, **common_mf_kws)
             raw_out = raw_sss
-            raw_fname_out = op.join(deriv_path, bids_basename + '_sss_raw.fif')
+            raw_fname_out = op.join(deriv_path, f'{bids_basename}_sss_raw.fif')
         else:
             msg = ('Not applying Maxwell filter.\nIf you wish to apply it, '
                    'set use_maxwell_filter=True in your configuration.')
-            logger.info(gen_log_message(message=msg, step=1,
-                                        subject=subject, session=session))
+            logger.info(gen_log_message(message=msg, step=1, subject=subject,
+                                        session=session))
             raw_out = raw
             raw_fname_out = op.join(deriv_path,
-                                    bids_basename + '_nosss_raw.fif')
-
-        os.makedirs(os.path.dirname(raw_fname_out), exist_ok=True)
-
+                                    f'{bids_basename}_nosss_raw.fif')
         raw_out.save(raw_fname_out, overwrite=True)
         if config.interactive:
             raw_out.plot(n_channels=50, butterfly=True)
 
+        # Empty-room processing.
+        #
+        # We pick the empty-room recording closest in time to the first run
+        # of the experimental session.
+        if run_idx == 0 and config.noise_cov == 'emptyroom':
+            msg = 'Processing empty-room recording …'
+            logger.info(gen_log_message(step=1, subject=subject,
+                                        session=session, message=msg))
+
+            bids_basename_er_in = get_matched_empty_room(
+                bids_basename=bids_basename,
+                bids_root=config.bids_root)
+            raw_er = load_data(bids_basename_er_in)
+            raw_er.info['bads'] = [ch for ch in raw.info['bads'] if
+                                   ch.startswith('MEG')]
+
+            # Maxwell-filter empty-room data.
+            if config.use_maxwell_filter:
+                msg = 'Applying Maxwell filter to empty-room recording'
+                logger.info(gen_log_message(message=msg, step=1,
+                                            subject=subject, session=session))
+
+                # We want to ensure we use the same coordinate frame origin in
```

Comment:
you mean coordinate frame transformation?

I don't see you setting the origin below

Reply:
The origin is inferred by `maxwell_filter()` via `mne.bem.fit_sphere_to_headshape()`, based on `info['dig']`

Reply:
I see, it wasn't completely clear from this comment. But now I understand